nsot/models/resource.py

```diff
@@ -69,40 +77,73 @@ def set_query(self, query, site_id=None, unique=False):
                 log.debug('Regex enabled for %r' % name)
 
             # Attribute lookup params
-            params = dict(
-                name=name, resource_name=resource_name
-            )
+            if name not in concrete_field_names:
+                params = dict(
+                    name=name, resource_name=resource_name
+                )
+            else:
+                params = dict()
+                params[name] = value
             # Only include site_id if it's set
             if site_id is not None:
                 params['site_id'] = site_id
 
             # If an Attribute doesn't exist, the set query is invalid. Return
             # an empty queryset. (fix #99)
-            try:
-                attr = Attribute.objects.get(
-                    **params
-                )
-            except Attribute.DoesNotExist as err:
-                raise exc.ValidationError({
-                    'query': '%s: %r' % (err.message.rstrip('.'), name)
-                })
+            if name in concrete_field_names:
+                try:
+                    # import pdb; pdb.set_trace()
+                    qfield = self.model.objects.filter(
```

Comment:
Here, you're storing `qfield` as a filtered list of objects based on a concrete field filter... But it doesn't appear this is every being used, since in the following block (line 118), `qfield` (`qfield.name` specifically) is only being used for non-concrete-field lookup.

Based on how this reads, it doesn't seem like `qfield` as model a queryset would ever make it past here, because if it did, `qfield.name` would likely result in an `AttributeError`.

It appears the only purpose of this initial filter query is to validate whether the concrete object fields exist in the first place so that the validation error can be raised. Can you please clarify?

Reply:
Yes you are correct un your understanding of the flow. I can add some comments here if you'd like?